mirage/utils.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable max-statements-per-line */
+import {Response} from 'ember-cli-mirage';
+
+export function paginatedResponse(modelName) {
+    return function (schema, request) {
+        let page = +request.queryParams.page || 1;
+        let limit = request.queryParams.limit || 15;
+        let pages, next, prev, models;
+
+        let allModels = this.serialize(schema[modelName].all())[modelName];
+
+        if (limit === 'all') {
+            pages = 1;
+        } else {
+            limit = +limit;
+
+            let start = (page - 1) * limit;
+            let end = start + limit;
+
+            pages = Math.ceil(allModels.length / limit);
+            models = allModels.slice(start, end);
+
+            if (start > 0) {
+                prev = page - 1;
+            }
+
+            if (end < allModels.length) {
+                next = page + 1;
+            }
+        }
+
+        return {
+            meta: {
+                pagination: {
+                    page,
+                    limit,
+                    pages,
+                    total: allModels.length,
+                    next: next || null,
+                    prev: prev || null
+                }
+            },
+            [modelName]: models || allModels
+        };
+    };
+}
+
+export function maintenanceResponse() {
+    return new Response(503, {}, {
+        errors: [{
+            errorType: 'Maintenance'
+        }]
+    });
+}
+
+export function versionMismatchResponse() {
+    return new Response(400, {}, {
+        errors: [{
+            errorType: 'VersionMismatchError'
+        }]
+    });
+}
+
+// minimalist deparam implementation
+export function deparam(string) {
```

Comment:
I'd say add it back in for all environments, it would be good to standardise on one library.

I'm aware that we have a hacky regex to parse a query param in [at least one place](https://github.com/TryGhost/Ghost-Admin/blob/lts/app/routes/setup.js#L31), it's probably worth an audit to make sure we're always using `deparam` when manually grabbing query parameters.